test/libsolidity/syntaxTests/inheritance/override/interface_and_base_override_fine.sol

```diff
@@ -0,0 +1,16 @@
+interface IBase {
+    function foo() external view;
+}
+
+contract Base is IBase {
+    function foo() public virtual view {}
+}
+
+interface IExt is IBase {}
+
+contract Ext is IExt, Base {}
+
+contract Impl is Ext {
+    function foo() public view override (IBase, Base) {}
```

Comment:
I rechecked now - and this is actually not a bug, it's actually how we specified this...
But nowadays, I'd actually change the spec and have this only require a single ``override`` here (resp. only require it to override ``Base``).

``Ext`` is fine even *without* overriding, since we have a unique choice of an implementation of ``foo``. That doesn't change in ``Impl``, so it's rather weird that we require *both* overrides there...

Reply:
I just remembered that this is probably coming from how I specified this originally: [Should overriding function declarations require the `override` keyword?](https://forum.soliditylang.org/t/should-overriding-function-declarations-require-the-override-keyword/347). And we were in favor of not relaxing this in the `override(A, B)` case. I.e. this is exactly what we wanted:

```solidity
interface IA {
    function foo() external;
}
interface IB {
    function foo() external;
}
contract C is IA, IB {
    function foo() external override(IA, IB) {}
}
```

Whether that was a good decision... given how we're ourselves confused by it now I'd say probably not :) Just dropping the requirement entirely would be more consistent. It does have one advantage though - you only have to treat interfaces specially in the single `override` case. In all other cases all inherited contracts count.

Reply:
Actually, this is not the whole story. If we did include the `override(A, B)` case for interfaces it would hide this problem but the problem itself is really what we have been discussing recently on the forum, just for interfaces. I.e. you need to include `IBase` because you're inheriting from it directly and not only through `Base`. The situation could be simplified to this:
```solidity
interface IBase {
    function foo() external view;
}
contract Base is IBase {
    function foo() public virtual view override {}
}
contract Ext is IBase, Base {
    function foo() public view virtual override(IBase, Base) {}
}
```
This requires `override(IBase, Base)` now but it did on 0.8.0 as well.

Reply:
After digging deeper into what's happening here, I think this is actually a bug. The message says that you need `override(Base, IBase)` here and IMO this would be correct in `Ext` but not in `Impl`. You should never have to specify more contracts than you directly inherit from. Each one you inherit from has to already have settled on a single implementation of `foo()` and you're overriding that specific implementation. I don't think it can just forward that decision down the inheritance hierarchy. At least not without making the intermediate contract `abstract`.

This is also how it worked before we relaxed the requirements for interfaces in #11628. I.e. it only required `override` (without specifying bases) on 0.8.0.

Reply:
Hm... In 0.8.0, it required to override ``IBase``, but *not* ``Base``... that was already wrong, but just less visibly wrong..